ext/standard/array.c

```diff
@@ -4639,7 +4639,7 @@ static void php_array_intersect_key(INTERNAL_FUNCTION_PARAMETERS, int data_compa
 
 	for (i = 0; i < argc; i++) {
 		if (Z_TYPE(args[i]) != IS_ARRAY) {
-			php_error_docref(NULL, E_WARNING, "Argument #%d is not an array", i + 1);
+			php_error_docref(NULL, E_WARNING, "Expected parameter %d to be an array, %s given", i + 1, zend_get_type_by_const(Z_TYPE(args[i])));
```

Comment:
Any reason to use zend_zval_type_name in some places and zend_get_type_by_const in others?

Reply:
I used `zend_zval_type_name` when possible as was the function I found for such a task. But, as it needs a `const zval *arg` I got an error type when using here, for e.g.:
```
expected ‘const zval * {aka const struct _zval_struct *}’ but argument is of type ‘zval {aka struct _zval_struct}’
```

So I search a way to retrieve the type, and come up with `zend_get_type_by_const` + `Z_TYPE`. Is there a better way? :)

Reply:
You can get a zval* from a zval by writing `&args[i]` instead of `args[i]`.